### PR TITLE
[DPP-701] Maybe missed error code conversions

### PR DESCRIPTION
**Pull request by pbatko-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala [hunk 1]
```diff
@@ -72,6 +72,7 @@ object ScenarioServiceMain extends App {
 
 object ScenarioService {
   private def notFoundContextError(id: Long): StatusRuntimeException =
+  // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:75]: Language team

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala [hunk 2]
```diff
@@ -292,8 +293,10 @@ class ScenarioService(implicit
 
         } catch {
           case e: archive.Error =>
+            // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:296]: Language team

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala [hunk 3]
```diff
@@ -292,8 +293,10 @@ class ScenarioService(implicit
...
             respObs.onError(Status.INVALID_ARGUMENT.withDescription(e.msg).asRuntimeException())
           case NonFatal(e) =>
+            // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:299]: Language team

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala [hunk 4]
```diff
@@ -439,6 +439,7 @@ class JsonLedgerClient(
         // TODO (MK) Using a grpc exception here doesn’t make that much sense.
         // We should refactor this to provide something more general.
         Future.successful(
+          // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala:442]: Language or runtime team?

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala [hunk 5]
```diff
@@ -55,6 +55,7 @@ final class TransactionsServiceImpl(ledgerContent: Observable[LedgerItem])
         Metadata.Key.of("cause", Metadata.ASCII_STRING_MARSHALLER),
         s"BEGIN should be strictly smaller than END. Found BEGIN '${request.getBegin}' and END '${request.getEnd}'",
       )
+      // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala:58]: Language team?

File: ledger/error/src/main/scala/com/daml/error/definitions/RejectionGenerators.scala [hunk 6]
```diff
@@ -191,6 +191,7 @@ class RejectionGenerators(conformanceMode: Boolean) {
   // TODO error codes: This converter is deprecated and should be removed
   //                   Instead of using this, construct proper validation errors in callers of this method
   //                   and only convert to StatusRuntimeExceptions when dispatched (e.g. in ApiSubmissionService)
+  @deprecated("not used in daml sdk")
```
Comment (tudor-da) [inline ledger/error/src/main/scala/com/daml/error/definitions/RejectionGenerators.scala:194]: 👍

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/BatchedValidatingCommitter.scala [hunk 7]
```diff
@@ -78,6 +78,7 @@ class BatchedValidatingCommitter[LogResult](
           Future.successful(SubmissionResult.Acknowledged)
         case Failure(exception) =>
           Future.successful(
+            // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/BatchedValidatingCommitter.scala:81]: KV Committer team

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/BatchingQueue.scala [hunk 8]
```diff
@@ -121,13 +121,15 @@ case class DefaultBatchingQueue(
                 )
               )
             case f: QueueOfferResult.Failure =>
+              // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/BatchingQueue.scala:124]: KV committer team

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/BatchingQueue.scala [hunk 9]
```diff
@@ -121,13 +121,15 @@ case class DefaultBatchingQueue(
...
               SubmissionResult.SynchronousError(
                 Status(
                   Code.INTERNAL.value,
                   f.toString,
                 )
               )
             case QueueOfferResult.QueueClosed =>
+              // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/BatchingQueue.scala:132]: KV committer team

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/ValidatingCommitter.scala [hunk 10]
```diff
@@ -65,13 +65,15 @@ class ValidatingCommitter[LogResult](
             postCommit(value)
             SubmissionResult.Acknowledged
           case Left(MissingInputState(keys)) =>
+            // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/ValidatingCommitter.scala:68]: KV committer team

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/ValidatingCommitter.scala [hunk 11]
```diff
@@ -65,13 +65,15 @@ class ValidatingCommitter[LogResult](
...
             SubmissionResult.SynchronousError(
               Status(
                 Code.INTERNAL.value,
                 s"Missing input state: ${keys.map(_.bytes.asScala.map("%02x".format(_)).mkString).mkString(", ")}",
               )
             )
           case Left(ValidationError(reason)) =>
+            // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/ValidatingCommitter.scala:76]: KV committer team

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala [hunk 12]
```diff
@@ -744,6 +744,7 @@ private[sandbox] final class InMemoryLedger(
       loggingContext: LoggingContext
   ): Future[Unit] =
     // sandbox-classic in-memory ledger does not support pruning
+  // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala:747]: Good find

File: ledger/sandbox-common/src/main/scala/platform/sandbox/services/SandboxResetService.scala [hunk 13]
```diff
@@ -44,6 +44,7 @@ class SandboxResetService(
       serverCallHandler: ServerCallHandler[ReqT, RespT],
   ): Listener[ReqT] = {
     if (resetInitialized.get) {
+      // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/sandbox-common/src/main/scala/platform/sandbox/services/SandboxResetService.scala:47]: Good find

File: ledger/sandbox-common/src/main/scala/platform/sandbox/services/SandboxResetService.scala [hunk 14]
```diff
@@ -71,6 +72,7 @@ class SandboxResetService(
     logger.info("Initiating server reset.")
 
     if (!resetInitialized.compareAndSet(false, true))
+    // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/sandbox-common/src/main/scala/platform/sandbox/services/SandboxResetService.scala:75]: Participant team

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadWriteServiceBridge.scala [hunk 15]
```diff
@@ -261,6 +261,7 @@ object ReadWriteServiceBridge {
           )
         case QueueOfferResult.Failure(throwable) =>
           logger.error("Error enqueueing new submission.", throwable)
+          // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadWriteServiceBridge.scala:264]: Participant team

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadWriteServiceBridge.scala [hunk 16]
```diff
@@ -269,6 +270,7 @@ object ReadWriteServiceBridge {
           )
         case QueueOfferResult.QueueClosed =>
           logger.error("Error enqueueing new submission: queue is closed.")
+          // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadWriteServiceBridge.scala:273]: Participant team

File: ledger/participant-integration-api/src/main/scala/platform/store/Conversions.scala [hunk 17]
```diff
@@ -332,6 +332,7 @@ private[platform] object Conversions {
     import RejectionReasonOps._
 
     def toParticipantStateRejectionReason: state.Update.CommandRejected.RejectionReasonTemplate =
+    // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/Conversions.scala:335]: Participant team: used in sandbox-classic

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/QueueBackedTracker.scala [hunk 18]
```diff
@@ -49,6 +49,7 @@ private[services] final class QueueBackedTracker(
             _.left.map(completionFailure => QueueCompletionFailure(completionFailure))
           )
         case QueueOfferResult.Failure(t) =>
+          // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/QueueBackedTracker.scala:52]: Participant team.

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala [hunk 19]
```diff
@@ -85,6 +85,7 @@ object CompletionResponse {
     }
 
   private[daml] def toException(response: TrackedCompletionFailure): StatusException =
+  // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:88]: Participant team

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala [hunk 20]
```diff
@@ -115,6 +116,7 @@ object CompletionResponse {
 
   private def buildException(metadata: Map[String, String], status: StatusJavaProto.Builder) = {
     val details = mergeDetails(metadata, status)
+    // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:119]: Participant team

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/SubmissionResult.scala [hunk 21]
```diff
@@ -28,6 +28,7 @@ object SubmissionResult {
     override val description: String = s"Submission failed with error ${status.message}"
 
     def exception: StatusRuntimeException =
+      // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/SubmissionResult.scala:31]: Here we are already getting a Status, expecting that the `WriteService` is returning self-service error code formatted statuses. It should already contain the self-service error code. I'll look into it, thanks 👍

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/Rejection.scala [hunk 22]
```diff
@@ -28,6 +28,7 @@ object Rejection {
       domain.RejectionReason.InvalidLedgerTime(description)
 
     override lazy val toStateRejectionReason: state.Update.CommandRejected.RejectionReasonTemplate =
+      // TODO error codes: Adapt V2 ?
```
Comment (tudor-da) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/Rejection.scala:31]: Participant team

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/QueueBackedTracker.scala (hunk 18 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/QueueBackedTracker.scala:52]: me

**Review by pbatko-da — COMMENTED** [review]
@fabiotudone-da, @remyhaemmerle-da, @stefanobaghino-da - Could you have a look? I mentioned you in the relevant comments.

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala (hunk 5 above)
Comment (mziolekda) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala:58]: This is actually for the runtime team

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala (hunk 19 above)
Comment (mziolekda) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala:88]: Just like all command-tracker stuff, it is incorporated both into the ledger-api-server and the client code. As such we need to keep the runtime team in the loop.
@stefanobaghino-da FYI

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala:442]: >    Language or runtime team?

Language is responsible from this part of the script. @stefanobaghino-da you can ignore this one.

**Comment by fabiotudone-da** [comment]
@pbatko-da We've decided not to port batching-related errors to V2 as now we only use pre-execution. CC @miklos-da

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala (hunk 5 above)
Comment (remyhaemmerle-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala:58]: This one is for you @stefanobaghino-da

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:75]: Could you be a bit more explicit about what you expect form Daml connect teams here ? 

Do you want to know if those error could happen ? If there is some missing ?

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:296]: Scenarios will die completely in Daml 2.0. 
Safe to ignore.

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:75]: scenarios will die completely, safe to ignore.

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala:299]: same here.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala:442]: cc @cocreature

Comment (cocreature) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/JsonLedgerClient.scala:442]: that’s fine, just keep it

**Comment by pbatko-da** [comment]
Thanks for the comments. To sum up:

1. Language sites are good to ignore (~going away in Daml 2.0)
2. Committer sites are good to ignore (as they batching-related which is not used)
3. Runtime site: https://github.com/digital-asset/daml/pull/11510#discussion_r742736025 waiting for @stefanobaghino-da  
4. Other sites are Participant

(FYI @remyhaemmerle-da @stefanobaghino-da @fabiotudone-da)

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala (hunk 5 above)
Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala:58]: Can you please create a ticket to track the required downstream changes that you are not in a position to address? Thanks.

Comment (mziolekda) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala:58]: https://github.com/digital-asset/daml/issues/11587

Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala:58]: For the record, this is just a mock, so no need to adapt this. This is not (and definitely _should_ not) be used to test responses to errors, so no need to have the mock implement self-service error codes. The important bits have already been handled as part of https://github.com/digital-asset/daml/pull/11223

**Pull request closed without merging**